python/ray/data/read_api.py

```diff
@@ -2668,6 +2668,98 @@ def from_arrow_refs(
     )
 
 
+@DeveloperAPI
+def read_delta_sharing_tables(
+    url: str,
+    limit: Optional[int] = None,
+    version: Optional[int] = None,
+    timestamp: Optional[str] = None,
+    jsonPredicateHints: Optional[str] = None,
+    parallelism: int = -1,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+    concurrency: Optional[int] = None,
+    override_num_blocks: Optional[int] = None,
+) -> Dataset:
+    """
+    Read data from a Delta Sharing table.
+
+    This function reads data from a Delta Sharing table specified by the URL.
+    It supports various options such as limiting the number of rows, specifying
+    a version or timestamp, and configuring parallelism and concurrency.
+
+    Before calling this function, ensure that the URL is correctly formatted
+    to point to the Delta Sharing table you want to access. Make sure you have
+    a valid delta_share profile in the working directory.
+
+    Examples:
+
+        .. testcode::
+            :skipif: True
+
+            import ray
+
+            ds = ray.data.read_delta_sharing_tables(
+                url=f"{current_address}/delta_share.json#share-with-coinbase-ray-cluster.\
+                        ml_adhoc_data_sharing.ralph_user_pairs_with_seq_features_310d_p01_train",
+                limit=100000,
+                version=1,
+            ).materialize()
+
+    Args:
+        url: A URL under the format "<profile>#<share>.<schema>.<table>".
```

Comment:
Do we have a link on Delta Sharing documentation for the URL format? If yes, we can add a link here.